rs/replicated_state/src/replicated_state.rs

```diff
@@ -99,12 +99,6 @@ pub enum StateError {
     /// Canister is stopping, only accepting responses.
     CanisterStopping(CanisterId),
 
-    /// Message enqueuing failed due to canister migration.
-    CanisterMigrating {
-        canister_id: CanisterId,
-        host_subnet: SubnetId,
-    },
```

Comment:
I removed this variant again because with `induct_message_impl()` returning a reject signal instead, this is no longer needed. It made sense before this PR because it allowed to unify the way reject responses were generated, but this kind of reject response is now gone altogether. The only reject responses we still have are locally generated upon observing a reject signal. The only way those get into a stream is during canister migration where they can't be inducted on the former host subnet.